### DISCUSSION TITLE
Ingest huge number of documents

### DISCUSSION THREAD

**Discussion by Oyaa404** [nextapps-de/flexsearch#319] [Q&A]
Q1 - I have a log file that has 500k of json objects, and each one is complex itself,
Any Idea for how to ingest this data with best performance? 
In a 30 seconds for example ?

Q2 - is there a way to push multiple document to the index, instead of pushing one by one?

**Comment by RupertBarrow** [comment]
It might be longer than that.
However, you can export the index, then reimport it every time you want to use it : that will be quicker.

However, the result might be huge and require a lot of memory.
A way of getting round this may be to design your search engine to break down a single index into multiple, smaller, indices which you can then search in parallel.
